### PR TITLE
GKE resource labels

### PR DESCRIPTION
**Pull request by jsirianni**
I have a use-case that requires resource labels on GKE for costing / billing filtering. This PR adds optional resource labels for GKE clusters. 

Default behavior is not modified. 

My tests passed when I ran `go test -v -timeout 60m -run ./gke_cluster_test.go`.

### CODE REVIEW

File: examples/gke-public-cluster/main.tf [hunk 1]
```diff
@@ -52,6 +52,10 @@ module "gke_cluster" {
   alternative_default_service_account = var.override_default_node_pool_service_account ? module.gke_service_account.email : null
 
   enable_vertical_pod_autoscaling = var.enable_vertical_pod_autoscaling
+
+  resource_labels = {
+    purpose = "testing"
```
Comment (robmorgan) [inline examples/gke-public-cluster/main.tf:57]:
```suggestion
    environment = "testing"
```
Let's change `purpose` to `environment` as it's likely a better example for most users.

**Review by robmorgan — COMMENTED** [review]
Hi @jsirianni,

Thanks for the PR! It looks good so far, I've added one minor suggestion. Also, would you mind rolling this out to all of the examples for consistency?

Cheers,

Rob

File: examples/gke-public-cluster/main.tf (hunk 1 above)
Comment (jsirianni) [reply examples/gke-public-cluster/main.tf]: I have pushed your change here: https://github.com/gruntwork-io/terraform-google-gke/pull/98/commits/4f14a759b0568252671448706c0668e0eff53d3c

**Comment by jsirianni** [comment]
Thanks for the quick turnaround @robmorgan. Please let me know if you need anything else.

**Comment by robmorgan** [comment]
@jsirianni LGTM!

**Review by robmorgan — APPROVED** [review]

**Pull request merged**